penguin/cli/opencode_launcher.py

```
@@ -39,6 +39,25 @@
 DEFAULT_TUI_RELEASE_URL = f"{DEFAULT_TUI_RELEASE_API_ROOT}/latest"
 DEFAULT_WEB_URL = f"http://127.0.0.1:{DEFAULT_WEB_PORT}"
 _URL_MODE_CAP_CACHE: dict[str, bool] = {}
+_STARTUP_PROFILE_START = time.perf_counter()
+
+
+def _profile_startup(label: str, **details: Any) -> None:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**details`

(ANN401)